cubi_tk/sodar/pull_data_collection.py

```diff
@@ -0,0 +1,367 @@
+        )
+
+        # Validate output directory path
+        if not os.path.exists(args.output_dir):
+            try:
+                os.makedirs(args.output_dir)
+            except Exception as e:
+                logger.error(
+                    f"Output directory path does not exist and cannot be created: {args.output_dir}"
+                )
+                logger.debug(e)
+                res = 1
+        elif not os.access(args.output_directory, os.W_OK):
+            logger.error(
+                f"Output directory path either does not exist or it is not writable: {args.base_path}"
+            )
+            res = 1
+
+        return res
+
+    def execute(self) -> typing.Optional[int]:
+        """Execute the transfer."""
+        res = self.check_args(self.args)
+        if res:  # pragma: nocover
+            return res
+
+        logger.info("Starting cubi-tk snappy pull-processed-data")
+        logger.info("  args: %s", self.args)
+
+        # Get list of sample ids
+        if self.args.sample_list:
+            samples = self.args.sample_list
+        elif self.args.biomedsheet:
+            samples = self.parse_sample_tsv(self.args.biomedsheet, sample_col=2, n_header_cols=13)
+        elif self.args.tsv:
+            samples = self.parse_sample_tsv(
+                self.args.tsv, sample_col=self.args.tsv_column, n_header_cols=self.args.tsv_header
+            )
+        else:
+            samples = None
+
+        # Find all remote files (iRODS)
+        filesearcher = RetrieveSodarCollection(
+            self.args.sodar_url,
+            self.args.sodar_api_token,
+            self.args.assay_uuid,
+            self.args.project_uuid,
+        )
+
+        remote_files_dict = filesearcher.perform()
+        assay_path = filesearcher.get_assay_irods_path(self.args.assay_uuid)
+
+        if self.args.all_files:
+            file_patterns = []
+        elif self.args.preset:
+            file_patterns = self.presets[self.args.preset]
+        else:  # self.args.file_pattern
+            file_patterns = self.args.file_pattern
+
+        filtered_remote_files_dict = self.filter_irods_file_list(
+            file_patterns, samples, self.args.substring_match, assay_path
+        )
+
+        if len(filtered_remote_files_dict) == 0:
+            self.report_no_file_found(available_files=[*remote_files_dict])
+            return 0
+
+        # Pair iRODS path with output path
+        transfer_jobs = self.build_download_jobs(
+            remote_files_dict=filtered_remote_files_dict,
+            assay_path=assay_path,
+        )
+
+        # Retrieve files from iRODS
+        iRODSTransfer(transfer_jobs).get(self.args.overwrite)
+
+        logger.info("All done. Have a nice day!")
+        return 0
+
+    @staticmethod
+    def parse_sample_tsv(tsv_path, sample_col=1, n_header_cols=1, skip_comments=True) -> List[str]:
+        samples = []
+        with open(tsv_path, "r") as f:
+            for _ in range(n_header_cols):
```

Comment:
I would prefer using csv library or any library that properly supports the format we are actually working with

Reply:
mmh, it seems the csv library doesn't support skipping commented lines (which we should have), so I don't think it's really woth using that.
We could go for pandas (which is already a cubi-tk dependency anyway), I'll look into that

Reply:
Switched to pandas for tsv parsing